tests/integration/api/mutations/navigationItem/deleteNavigationItem.test.js

```diff
@@ -0,0 +1,113 @@
+import importAsString from "@reactioncommerce/api-utils/importAsString.js";
+import decodeOpaqueIdForNamespace from "@reactioncommerce/api-utils/decodeOpaqueIdForNamespace.js";
+import Factory from "/tests/util/factory.js";
+import TestApp from "/tests/util/TestApp.js";
+
+const CreateNavigationItemMutation = importAsString("./CreateNavigationItemMutation.graphql");
+const DeleteNavigationItemMutation = importAsString("./DeleteNavigationItemMutation.graphql");
+
+jest.setTimeout(300000);
+
+const internalShopId = "123";
+const opaqueShopId = "cmVhY3Rpb24vc2hvcDoxMjM="; // reaction/shop:123
+const shopName = "Test Shop";
+
+const mockAdminAccount = Factory.Account.makeOne({
+  roles: {
+    [internalShopId]: ["admin", "core"]
+  }
+});
+
+let testApp;
+let createNavigationItem;
+let deleteNavigationItem;
+let createdNavigationItemOpaqueId;
+
+beforeAll(async () => {
+  testApp = new TestApp();
+  await testApp.start();
+  await testApp.insertPrimaryShop({ _id: internalShopId, name: shopName });
+  await testApp.createUserAndAccount(mockAdminAccount);
+  await testApp.setLoggedInUser(mockAdminAccount);
+
+  createNavigationItem = testApp.mutate(CreateNavigationItemMutation);
+  deleteNavigationItem = testApp.mutate(DeleteNavigationItemMutation);
+
+  const { createNavigationItem: createResult } = await createNavigationItem({
```

Comment:
@mikemurray I think we should try to keep tests as isolated as possible from other APIs as if `CreateNavigationItemMutation` fails it will lead to failure of unrelated `deleteNavigationItem` and `updateNavigationItem` tests. 

So for every `CreateNavigationItemMutation` input/payload update, we will need to make sure all dependent tests are updates. 

I might be missing some point here as we are not creating data using factory like other tests, so please feel free to shed some light.

Reply:
The create/update/delete mutations are all part of the navigation API so if the create is broken, then the delete and update especially could be affected.

We have no rules that state that integration tests must use factories, nor if they can or cannot use queries/mutations from other parts of the API where available. We are trying to verify things work together in full app integration, not as individual units. That being said, I would avoid going out of the boundaries of the plugin you're testing (unless it's necessary for the test). At that point, factories and mocks may prove useful.

Reply:
understood @mikemurray , really appreciate the quick explanation. Also, this is an "integration test" 😄 .  This will help me write better tests in the future.